### DISCUSSION TITLE
default=[] and default_factory=list marks field as `Optional` in OpenAPI documentation

### DISCUSSION THREAD

**Discussion by JonasKs** [pydantic/pydantic#4577] [Question] [answered]
Hi,

When I have a model like this:

```python
class X(BaseModel):
    list_item: list[str] = Field(default=[])
```

The list will always be there, it won't be `None`. I have also not type annotated it as `list[str] | None`. However, the OpenAPI documentation does not mark it as required. 

The solution is to do this:
```python
class X(BaseModel):
    list_item: list[str] = Field(default=...)
```
but that also forces me to always create `X(list_item=[])` instead of just `X()`. Not the end of the world, but for some large models this is a bit annoying.

Is OpenAPI documentation generated without using type hints, since this don't work?   
Any clean workarounds?

**Comment by JonasKs** [comment]
I guess a decent workaround is to do this:
```python
class X(BaseModel):
    list_item: list[str] = Field(default=[])

    class Config:
        schema_extra = {
           'required': ['list_item']
        }
```
Even though the behavior seems a bit counterintuitive to me.

**Comment by samuelcolvin** [comment, marked as answer]
Moved to issue and replied there, see #4697.
